src/Linters/ShoutCaseEnumMembersLinter.hack

```diff
@@ -0,0 +1,111 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+use namespace HH\Lib\{C, Dict, Regex, Str, Vec};
+
+final class ShoutCaseEnumMembersLinter extends AutoFixingASTLinter {
+  const type TContext = EnumDeclaration;
+  const type TNode = Enumerator;
+
+  <<__Override>>
+  public function getLintErrorForNode(
+    self::TContext $enum,
+    self::TNode $member,
+  ): ?ASTLintError {
+    $original_name = $this->memberToName($member);
+    if ($this->isShoutCase($original_name)) {
+      return null;
+    }
+
+    $transformed_names = $this->prepareAllNewNamesFor($enum);
+
+    // If we can't find the shout name, we bail, because the enum
+    // member names would not be unique after the fix.
+    if (!C\contains_key($transformed_names, $original_name)) {
+      return new ASTLintError(
+        $this,
+        Str\format(
+          'Member {%s} is not in SHOUT_CASE,
+but the autogenerated name collided, so no autofix is suggested.',
+          $original_name,
+        ),
+        $member,
+        null,
+      );
+    }
+
+    return new ASTLintError(
+      $this,
+      Str\format('Member {%s} is not in SHOUT_CASE', $original_name),
+      $member,
+      () ==> $this->getFixedNode($member),
```

Comment:
I think just

```hack
() ==> $node->withName($node->getName()->withText($transformed_names[$original_name]))
```

might be more straightforward (also saves the extra call to transformToShoutCase)

It also feels a bit weird to create that dict and only ever use it for contains_key :) -- in that case it could've just been a keyset -- but again, we do need the value, so let's get it from there.